### ISSUE TITLE
Allow ProductAPI resources to be generated without specifying products array in yml file

### ISSUE THREAD

**Issue by levimatheri** [mirsaeedi/dotnet-apim#37]
First of all, I love this tool! It would be nice if we didn't have to specify the `products` array in the yml file for the product-api assignments to be added to the output ARM template. This is useful because we manage products as separate arm templates from the APIs. This is the behavior in the apim-azure-devops-toolkit, as in [this section of code](https://github.com/Azure/azure-api-management-devops-resource-kit/blob/a993488b807459713ee991e36cfbbd5d2f340d42/src/ArmTemplates/Creator/TemplateCreators/ProductAPITemplateCreator.cs#L47), it only adds `dependsOn` on `products/apis` resource and not `products`. Besides, we're already adding `dependsOn` `product` in the API resource, i.e. products is added as a [dependency in `ApiDeploymentDefinition`](https://github.com/mirsaeedi/dotnet-apim/blob/e5267906b1024cbd995d6913b741f87bf9dced38/src/apimtemplate/Core/DeploymentDefinitions/Entities/ApiDeploymentDefinition.cs#L161-L166). Therefore, if one decides to add the products array in their yml, it should still work.

Currently, if one doesn't include `products` in the yml, deployment fails with this error: _"The resource 'Microsoft.ApiManagement/service/products/***' is not defined in the template"_

@mirsaeedi I'd be happy to submit a PR for this. 

Thanks.

**Comment by mirsaeedi** [comment]
Hello @levimatheri. Thanks for you feedback :) 
Can you please elaborate more on this issue? the `products` array is not mandatory in the yml file 🤔

**Comment by levimatheri** [comment]
Hi @mirsaeedi. Yes, the `products` array is not required in the yml, but if we don't include it and add product assignments in the API definition, the generated ARM adds a `dependsOn` the product. Since the product resource isn't included in the generated ARM (i.e. because of omitting the `products` array in the yml), the deployment fails with the error I indicated above, because `dependsOn` requires that the referenced resource be in the same ARM template. This is the case with API Tags as well. I've updated the PR.

See example:
Here's my yml:
![image](https://user-images.githubusercontent.com/16405269/177232158-b295cac6-de4d-4faa-b180-f4df49bfc82d.png)


Here's the generated ARM template:
![image](https://user-images.githubusercontent.com/16405269/177230536-5e43b00a-78e1-4f37-9c6f-31ca3cbac6b7.png)

This is what I'd expect instead:
![image](https://user-images.githubusercontent.com/16405269/177232294-79e70609-fc59-4280-aed7-a83b38366020.png)
